uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -0,0 +1,492 @@
+define([
+  'leave-absences/shared/modules/controllers',
+  'common/lodash',
+  'common/moment',
+  'common/services/api/option-group',
+  'common/services/hr-settings',
+], function (components, _, moment) {
+  'use strict';
+
+  components.controller('LeaveRequestPopupCtrl', [
+    '$log', '$q', '$uibModalInstance', 'AbsencePeriod', 'AbsenceType', 'Entitlement',
+    'Calendar', 'LeaveRequestInstance', 'LeaveRequest', 'api.optionGroup', 'baseData',
+    'PublicHoliday', 'HR_settings', '$rootScope',
+    function ($log, $q, $modalInstance, AbsencePeriod, AbsenceType, Entitlement,
+      Calendar, LeaveRequestInstance, LeaveRequest, OptionGroup, baseData, PublicHoliday,
+      HR_settings, $rootScope) {
+      $log.debug('LeaveRequestPopupCtrl');
+
+      var $ctrl = this,
```

Comment:
Let's use the same `vm` pattern
```js
var vm = {};
```
Do you need it to be equal to `this`? Is there any property that is automatically attached to `this` that you need to use? If no, use the above ^, if yes, you can do
```js
var vm = Object.create(this);
```